zppy/ilamb_run.py

```diff
@@ -0,0 +1,92 @@
+import os
+import pprint
+
+import jinja2
+
+from zppy.utils import checkStatus, getTasks, getYears, submitScript
+
+
+# -----------------------------------------------------------------------------
+def ilamb_run(config, scriptDir):
+
+    # Initialize jinja2 template engine
+    templateLoader = jinja2.FileSystemLoader(
+        searchpath=config["default"]["templateDir"]
+    )
+    templateEnv = jinja2.Environment(loader=templateLoader)
+    template = templateEnv.get_template("ilamb_run.bash")
+
+    # --- List of ilamb_run tasks ---
+    tasks = getTasks(config, "ilamb_run")
+    if len(tasks) == 0:
+        return
+
+    # --- Generate and submit ilamb_run scripts ---
+    dependencies = []
+
+    for c in tasks:
+
+        if "ts_num_years" in c.keys():
+            c["ts_num_years"] = int(c["ts_num_years"])
+
+        # Loop over year sets
+        year_sets = getYears(c["years"])
+        for s in year_sets:
+            c["year1"] = s[0]
+            c["year2"] = s[1]
+            c["scriptDir"] = scriptDir
+
+            # List of dependencies
+            dependencies.append(
+                os.path.join(
+                    scriptDir,
+                    "ts_%s_%04d-%04d-%04d.status"
+                    % ("land_monthly", c["year1"], c["year2"], c["ts_num_years"]),
```

Comment:
Is "land_monthly" always going to be the subsection?

Reply:
I meant to have "land_monthly" the minimum dependency. One can have "atm_monthly" data processed, but not required.